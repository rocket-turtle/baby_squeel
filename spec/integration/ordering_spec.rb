require 'spec_helper'

describe '#ordering' do
  it 'orders using arel' do
    relation = Post.ordering { title.desc }
    expect(relation).to match_sql_snapshot
  end

  it 'orders using multiple arel columns' do
    relation = Post.ordering {
      [title.desc, published_at.asc]
    }

    expect(relation).to match_sql_snapshot
  end

  it 'orders on an aggregates' do
    relation = Post.group('"posts"."id"').ordering {
      id.count.desc
    }

    expect(relation).to match_sql_snapshot
  end

  it 'orders using operations' do
    relation = Post.joins(:author).ordering {
      (author.id - id).desc
    }

    expect(relation).to match_sql_snapshot
  end

  it 'orders on an aliased table' do
    relation = Post.joins(author: :posts).ordering {
      author.posts.id
    }

    expect(relation).to match_sql_snapshot
  end
end
