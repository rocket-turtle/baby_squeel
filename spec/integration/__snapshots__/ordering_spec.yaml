---
"#ordering orders using arel 1": SELECT "posts".* FROM "posts" ORDER BY "posts"."title"
  DESC
"#ordering orders using multiple arel columns 1": SELECT "posts".* FROM "posts" ORDER
  BY "posts"."title" DESC, "posts"."published_at" ASC
"#ordering orders on an aggregates 1": SELECT "posts".* FROM "posts" GROUP BY "posts"."id"
  ORDER BY COUNT("posts"."id") DESC
"#ordering orders using functions 1": SELECT "posts".* FROM "posts" INNER JOIN "authors"
  ON "authors"."id" = "posts"."author_id" ORDER BY coalesce("posts"."id", "authors"."id")
  DESC
"#ordering orders using operations 1": SELECT "posts".* FROM "posts" INNER JOIN "authors"
  ON "authors"."id" = "posts"."author_id" ORDER BY ("authors"."id" - "posts"."id")
  DESC
"#ordering orders on an aliased table 1": SELECT "posts".* FROM "posts" INNER JOIN
  "authors" ON "authors"."id" = "posts"."author_id" INNER JOIN "posts" "posts_authors"
  ON "posts_authors"."author_id" = "authors"."id" ORDER BY "posts_authors"."id"
"#ordering orders on an aliased table 1 (Active Record: v8.1)": SELECT "posts".* FROM
  "posts" INNER JOIN "authors" ON "authors"."id" = "posts"."author_id" INNER JOIN
  "posts" AS "posts_authors" ON "posts_authors"."author_id" = "authors"."id" ORDER
  BY "posts_authors"."id"
